### PR TITLE
Fix default output-dir

### PR DESCRIPTION
**Pull request by anjastrunk**
While trying to generate credentials for CSPs with

```
python3 -m generator.cli csp
```

I received the following unexpected error:

```
Traceback (most recent call last):
  File "/home/anja-strunk/Repos/SCS/gx-credential-generator/generator/cli.py", line 363, in <module>
    cli_commands()
  File "/home/anja-strunk/VirtualEnv/scs/gx-credential-generator/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/anja-strunk/VirtualEnv/scs/gx-credential-generator/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/anja-strunk/VirtualEnv/scs/gx-credential-generator/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/anja-strunk/VirtualEnv/scs/gx-credential-generator/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/anja-strunk/VirtualEnv/scs/gx-credential-generator/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/anja-strunk/Repos/SCS/gx-credential-generator/generator/cli.py", line 88, in wrapper
    return func(*args, **kwargs)
  File "/home/anja-strunk/Repos/SCS/gx-credential-generator/generator/cli.py", line 196, in csp
    _print_vcs(vcs, out_dir)
  File "/home/anja-strunk/Repos/SCS/gx-credential-generator/generator/cli.py", line 308, in _print_vcs
    if not os.path.isdir(out_dir):
  File "/usr/lib/python3.10/genericpath.py", line 42, in isdir
    st = os.stat(s)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
```

The problem was, missing `out-dir` value for command `csp`.

A further bug was wrong formatted request to compliance service, resulting in an HTTP 409.

### CODE REVIEW

File: generator/discovery/csp_generator.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def generate(self) -> dict:
         # request Gaia-X compliance credential for CSP as Legal Person
         print('Request VC of type "gx:compliance" for CSP at GXDCH Compliance Service...', end='')
         vp = credentials.convert_to_vp(creds=[tandc_vc, lrn_vc, lp_vc])
-        cs_vc = self.compliance.request_compliance_vc(vp, self.cred_base_url + "/csp_compliance.json")
+        cs_vc = self.compliance.request_compliance_vc(json.dumps(vp), self.cred_base_url + "/csp_compliance.json")
```
Comment (markus-hentsch) [inline generator/discovery/csp_generator.py:59]: If I test using this change, the Gaia-X API crashes:

```
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url:
https://compliance.lab.gaia-x.eu/v1-staging/api/credential-offers?vcid=https://gaia-x-testing.cloudandheat.com/csp_compliance.json
Request VC of type "gx:compliance" for CSP at GXDCH Compliance Service...
```

If I revert this change, the request succeeds.

**Review by markus-hentsch — CHANGES_REQUESTED** [review]
I'm unable to reproduce the 409 error you mentioned. Actually your change to fix it breaks the process for me instead. Can you double check on your side?

File: generator/discovery/csp_generator.py (hunk 1 above)
Comment (anjastrunk) [reply generator/discovery/csp_generator.py]: You are right, there was a `json.dumps()` too much. I removed it, and generation should work now.

**Comment by markus-hentsch** [comment]
I attempted to retest this again after the breaking change was reverted. This time however, I ran into the 409 error myself.

At first I had trouble figuring out the cause due to missing error response details, which I attempt to address with #118 now. Being able to see the error message, it turns out that my Let's Encrypt certificate of my DID test server had expired over the recent days.

I went the long route and started to re-issue a certificate, rebuild and replace the DID as well as the hosted certificate chain and signed credentials.

However, no matter how often I repeated the process, the Gaia-X API would still refuse to validate the PEM file linked in the `did.json`. It took me a long time to debug. I eventually opened up my fullchain.pem in [KeyStore Explorer](https://keystore-explorer.org/) and checked against a copy of the old one. Then I realized what I already had written in the blog post but totally forgot about: the Let's Encrypt certbot's produced `fullchain.pem` is not the *full* chain. It's only the leaf and intermediate certificate but is missing the [Let's Encrypt's root certificate](https://letsencrypt.org/certificates/#root-cas). Browsers will accept this but tools that require the full chain locally won't, e.g., the Gaia-X servers.

After I built the new certificate chain correctly, everything goes smoothly again and I can finally approve this.

**Review by markus-hentsch — APPROVED** [review]
LGTM

**Pull request merged**